runtime/queries/column_timeseries.go

```diff
@@ -63,43 +64,40 @@ func (q *ColumnTimeseries) Resolve(ctx context.Context, rt *runtime.Runtime, ins
 		return fmt.Errorf("not available for dialect '%s'", olap.Dialect())
 	}
 
-	timeRange, err := q.normaliseTimeRange(ctx, rt, instanceID, priority)
+	timeRange, err := q.resolveNormaliseTimeRange(ctx, rt, instanceID, priority)
 	if err != nil {
 		return err
 	}
+
 	if timeRange.Interval == runtimev1.TimeGrain_TIME_GRAIN_UNSPECIFIED {
 		q.Result = &runtimev1.TimeSeriesResponse{}
 		return nil
 	}
+
 	var measures = normaliseMeasures(q.Measures, true)
-	var timestampColumn = q.TimestampColumnName
-	var tableName = q.TableName
-	var filter string
-	if q.Filters != nil {
-		filter = getFilterFromMetricsViewFilters(q.Filters)
-	}
-	var timeGranularity = convertToDateTruncSpecifier(timeRange.Interval)
-	tsAlias := "_ts_" + ReplaceHyphen(uuid.New().String())
+	filter := getFilterFromMetricsViewFilters(q.Filters)
+	dateTruncSpecifier := convertToDateTruncSpecifier(timeRange.Interval)
+	tsAlias := tempName("_ts_")
 	if filter != "" {
 		filter = "WHERE " + filter
 	}
-	temporaryTableName := "_timeseries_" + uuid.New().String()
-	sql := `CREATE TEMPORARY TABLE "` + temporaryTableName + `" AS (
+	temporaryTableName := tempName("_timeseries_")
+	sql := `CREATE TEMPORARY TABLE ` + safeName(temporaryTableName) + ` AS (
         -- generate a time series column that has the intended range
         WITH template as (
           SELECT 
             generate_series as ` + tsAlias + `
           FROM 
             generate_series(
-              date_trunc('` + timeGranularity + `', TIMESTAMP '` + timeRange.Start.AsTime().Format(IsoFormat) + `'),
-              date_trunc('` + timeGranularity + `', TIMESTAMP '` + timeRange.End.AsTime().Format(IsoFormat) + `'),
-              interval '1 ` + timeGranularity + `')
+              date_trunc('` + dateTruncSpecifier + `', TIMESTAMP '` + timeRange.Start.AsTime().Format(IsoFormat) + `'),
+              date_trunc('` + dateTruncSpecifier + `', TIMESTAMP '` + timeRange.End.AsTime().Format(IsoFormat) + `'),
+              interval '1 ` + dateTruncSpecifier + `')
         ),
         -- transform the original data, and optionally sample it.
         series AS (
           SELECT 
-            date_trunc('` + timeGranularity + `', "` + EscapeDoubleQuotes(timestampColumn) + `") as ` + tsAlias + `,` + getExpressionColumnsFromMeasures(measures) + `
-          FROM "` + EscapeDoubleQuotes(tableName) + `" ` + filter + `
+            date_trunc('` + dateTruncSpecifier + `', ` + safeName(q.TimestampColumnName) + `) as ` + tsAlias + `,` + getExpressionColumnsFromMeasures(measures) + `
+          FROM ` + safeName(q.TableName) + ` ` + filter + `
```

Comment:
In `getExpressionColumnsFromMeasures`, the `measure.SqlName` should be wrapped with `safeName`